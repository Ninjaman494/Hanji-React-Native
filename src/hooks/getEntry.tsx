import { gql, QueryResult, useQuery } from "@apollo/client";

const ENTRY = gql`
  query EntryQuery($id: ID!) {
    entry(id: $id) {
      id
      term
      pos
      definitions
      antonyms
      synonyms
      examples {
        sentence
        translation
      }
      regular
      note
    }
  }
`;

type Entry = {
  id: string;
  term: string;
  pos: string;
  definitions: string[];
  antonyms?: string[];
  synonyms?: string[];
  examples?: {
    sentence: string;
    translation: string;
  }[];
  regular?: boolean;
  note?: string;
};

type GetEntryResponse = {
  entry: Entry;
};

export default function getEntry(id: string): QueryResult<GetEntryResponse> {
  return useQuery<GetEntryResponse>(ENTRY, { variables: { id: id } });
}
