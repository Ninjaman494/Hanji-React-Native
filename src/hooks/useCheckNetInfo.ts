import NetInfo, { NetInfoState } from "@react-native-community/netinfo";
import { useEffect, useState } from "react";

const useCheckNetInfo = () => {
  const [netInfo, setNetInfo] = useState<NetInfoState>();

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener((state) => {
      console.log("Connection type", state.type);
      console.log("Is connected?", state.isInternetReachable);
      setNetInfo(state);
    });
    return unsubscribe;
  });

  return netInfo;
};

export default useCheckNetInfo;
