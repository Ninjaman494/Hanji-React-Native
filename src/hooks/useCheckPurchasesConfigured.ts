import { useEffect, useState } from "react";
import Purchases from "react-native-purchases";

const MINUTE_MS = 1000;

/**
 * Hook that checks if RevenueCat's Purchases object has been
 * configured.It will check every minute until it has been
 * configured.
 *
 * @returns true if RevenueCat's Purchases object is configured,
 * false otherwise
 */
const useCheckPurchasesConfigured = () => {
  const [isConfigured, setConfigured] = useState(false);

  useEffect(() => {
    const interval = setInterval(async () => {
      const configured = await Purchases.isConfigured();
      setConfigured(configured);

      if (configured) clearInterval(interval);
    }, MINUTE_MS);

    return () => clearInterval(interval);
  }, [setConfigured]);

  return isConfigured;
};

export default useCheckPurchasesConfigured;
