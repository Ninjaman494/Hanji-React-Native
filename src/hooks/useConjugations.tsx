import { gql, QueryHookOptions, useQuery } from "@apollo/client";

const CONJUGATIONS = gql`
  query ConjugationsQuery(
    $stem: String!
    $isAdj: Boolean!
    $honorific: Boolean!
    $regular: Boolean
    $conjugations: [String]
  ) {
    conjugations(
      stem: $stem
      isAdj: $isAdj
      honorific: $honorific
      regular: $regular
      conjugations: $conjugations
    ) {
      name
      conjugation
      type
      tense
      speechLevel
      honorific
      pronunciation
      romanization
      reasons
    }
  }
`;

export type Conjugation = {
  name: string;
  conjugation: string;
  type: string;
  tense: "PRESENT" | "PAST" | "FUTURE" | "NONE";
  speechLevel:
    | "INFORMAL_LOW"
    | "INFORMAL_HIGH"
    | "FORMAL_LOW"
    | "FORMAL_HIGH"
    | "NONE";
  honorific: boolean;
  pronunciation: string;
  romanization: string;
  reasons: string[];
};

export interface UseConjugationsVars {
  stem: string;
  isAdj: boolean;
  honorific: boolean;
  regular?: boolean;
  conjugations?: string;
}

type ConjugationsResponse = {
  conjugations: Conjugation[];
};

export interface UseConjugationsVars {
  stem: string;
  isAdj: boolean;
  honorific: boolean;
  regular?: boolean;
  conjugations?: string;
}

const useConjugations = (
  { stem, isAdj, honorific, regular, conjugations }: UseConjugationsVars,
  options?: QueryHookOptions
) => {
  return useQuery<ConjugationsResponse>(CONJUGATIONS, {
    ...options,
    variables: {
      stem: stem,
      isAdj: isAdj,
      honorific: honorific,
      regular: regular,
      conjugations: conjugations,
    },
  });
};

export default useConjugations;
