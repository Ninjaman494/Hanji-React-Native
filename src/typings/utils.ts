export type FormikContext = Record<
  string,
  string | number | boolean | Record<string, unknown>
>;
