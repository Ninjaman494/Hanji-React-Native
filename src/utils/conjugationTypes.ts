export enum Tense {
  PRESENT,
  PAST,
  FUTURE,
  NONE,
}

export enum Formality {
  INFORMAL_LOW = "informal low",
  INFORMAL_HIGH = "informal high",
  FORMAL_LOW = "formal low",
  FORMAL_HIGH = "formal high",
  NONE = "none",
}

export enum ConjugationType {
  CONNECTIVE_IF = "connective if",
  CONNECTIVE_AND = "connective and",
  CONNECTIVE_BUT = "connective but",
  DECLARATIVE_PRESENT = "declarative present",
  DECLARATIVE_PAST = "declarative past",
  DECLARATIVE_FUTURE = "declarative future",
  DETERMINER_PRESENT = "determiner present",
  DETERMINER_PAST = "determiner past",
  DETERMINER_FUTURE = "determiner future",
  IMPERATIVE = "imperative",
  INTERROGATIVE_PRESENT = "interrogative present",
  INTERROGATIVE_PAST = "interrogative past",
  NOMINAL_ING = "nominal ing",
  PROPOSITIVE = "propositive",
  SUPPOSITIVE = "suppositive",
}

type FormalityMinusNone = Exclude<Formality, Formality.NONE>;

export type ConjugationName =
  | ConjugationType.CONNECTIVE_IF
  | ConjugationType.CONNECTIVE_AND
  | ConjugationType.CONNECTIVE_BUT
  | ConjugationType.NOMINAL_ING
  | ConjugationType.DETERMINER_PRESENT
  | ConjugationType.DETERMINER_PAST
  | ConjugationType.DETERMINER_FUTURE
  | `${ConjugationType.DECLARATIVE_PRESENT} ${FormalityMinusNone}`
  | `${ConjugationType.DECLARATIVE_PAST} ${FormalityMinusNone}`
  | `${ConjugationType.DECLARATIVE_FUTURE} ${FormalityMinusNone}`
  | `${ConjugationType.IMPERATIVE} ${FormalityMinusNone}`
  | `${ConjugationType.INTERROGATIVE_PRESENT} ${FormalityMinusNone}`
  | `${ConjugationType.INTERROGATIVE_PAST} ${FormalityMinusNone}`
  | `${ConjugationType.PROPOSITIVE} ${FormalityMinusNone}`
  | `${ConjugationType.SUPPOSITIVE} ${FormalityMinusNone}`;
