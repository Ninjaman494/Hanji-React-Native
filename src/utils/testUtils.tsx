import {
  render,
  RenderAPI,
  RenderOptions,
} from "@testing-library/react-native";
import React, { FC, PropsWithChildren, ReactElement } from "react";
import { Provider as PaperProvider } from "react-native-paper";
import theme from "theme";

const Providers: FC<PropsWithChildren> = ({ children }) => {
  return <PaperProvider theme={theme}>{children}</PaperProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderAPI => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
