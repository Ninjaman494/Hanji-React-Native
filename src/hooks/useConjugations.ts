import { gql, QueryHookOptions, QueryResult, useQuery } from "@apollo/client";
import { useEffect, useState } from "react";
import { ConjugationName, Formality, Tense } from "../utils/conjugationTypes";

const CONJUGATIONS = gql`
  query GetConjugations($input: ConjugationsInput!) {
    getConjugations(input: $input) {
      name
      conjugation
      type
      tense
      speechLevel
      honorific
      pronunciation
      romanization
      reasons
    }
  }
`;

export type Conjugation = {
  name: ConjugationName;
  conjugation: string;
  type: string;
  tense: Tense;
  speechLevel: Formality;
  honorific: boolean;
  pronunciation: string;
  romanization: string;
  reasons: string[];
};

export interface UseConjugationsVars {
  input: {
    stem: string;
    isAdj: boolean;
    honorific?: boolean;
    regular: boolean | undefined; // If an entry has regular it MUST be passed in
    conjugations?: {
      name: string;
      honorific: boolean;
    }[];
  };
}

type ConjugationsResponse = {
  getConjugations?: Conjugation[];
};

const useConjugations = (
  vars: UseConjugationsVars,
  options?: QueryHookOptions
): QueryResult<ConjugationsResponse> => {
  const { data, ...rest } = useQuery<ConjugationsResponse>(CONJUGATIONS, {
    ...options,
    variables: vars,
  });

  const [conjugations, setConjugations] = useState<Conjugation[] | undefined>(
    undefined
  );

  useEffect(() => {
    if (data) {
      setConjugations(
        data.getConjugations?.map((c) => ({
          ...c,
          speechLevel: c.speechLevel
            .toLowerCase()
            .split("_")
            .join(" ") as Formality,
        }))
      );
    }
  }, [data, setConjugations]);

  return { data: { getConjugations: conjugations }, ...rest };
};

export default useConjugations;
