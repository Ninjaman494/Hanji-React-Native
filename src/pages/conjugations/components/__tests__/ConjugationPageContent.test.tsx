import { Conjugation } from "hooks/useConjugations";
import React from "react";
import "react-native";
import { ConjugationName, Formality, Tense } from "utils/conjugationTypes";
import { render } from "utils/testUtils";
import ConjugationsPageContent from "../ConjugationPageContent";

describe("ConjugationPageContent component", () => {
  const typeOneBase = {
    name: "name 1" as ConjugationName,
    conjugation: "conj 1",
    type: "declarative present",
    tense: Tense.PRESENT,
    speechLevel: Formality.INFORMAL_LOW,
    honorific: false,
    pronunciation: "pronunciation",
    romanization: "romanization",
    reasons: [],
  };

  const typeOneConjugations: Conjugation[] = [
    typeOneBase,
    {
      ...typeOneBase,
      speechLevel: Formality.INFORMAL_HIGH,
      conjugation: "conj 2",
    },
    {
      ...typeOneBase,
      speechLevel: Formality.FORMAL_LOW,
      conjugation: "conj 3",
    },
  ];

  const typeTwoBase = {
    name: "name 1" as ConjugationName,
    conjugation: "conj 3",
    type: "declarative future",
    tense: Tense.PRESENT,
    speechLevel: Formality.INFORMAL_LOW,
    honorific: false,
    pronunciation: "pronunciation",
    romanization: "romanization",
    reasons: [],
  };

  const typeTwoConjugations: Conjugation[] = [
    typeTwoBase,
    {
      ...typeTwoBase,
      speechLevel: Formality.INFORMAL_HIGH,
      conjugation: "conj 4",
    },
    {
      ...typeTwoBase,
      speechLevel: Formality.FORMAL_LOW,
      conjugation: "conj 5",
    },
  ];

  const props = {
    conjugations: typeOneConjugations.concat(typeTwoConjugations),
    onScroll: jest.fn(),
  };

  it("groups conjugations based on type", () => {
    const component = render(<ConjugationsPageContent {...props} />);

    expect(component).toMatchSnapshot();
  });
});
