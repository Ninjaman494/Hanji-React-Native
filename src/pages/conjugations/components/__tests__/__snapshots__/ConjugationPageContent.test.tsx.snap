// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConjugationPageContent component groups conjugations based on type 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    collapsable={false}
    data={
      [
        "declarative present",
        "declarative future",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={4}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={{}}
    viewabilityConfigCallbackPairs={[]}
    windowSize={5}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          collapsable={false}
          onLayout={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(0, 0, 0, 0.12)",
              "borderRadius": 10,
              "elevation": 5,
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 5,
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                  },
                  {
                    "minHeight": 50,
                  },
                  {},
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "500",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontSize": 20,
                            "letterSpacing": 0.15,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          [
                            {
                              "minHeight": 30,
                              "paddingRight": 16,
                            },
                            {
                              "marginBottom": 2,
                            },
                            {
                              "color": "#3F51B5",
                              "fontSize": 16,
                              "fontWeight": "400",
                            },
                          ],
                        ],
                      ],
                    ]
                  }
                >
                  Declarative Present
                </Text>
              </View>
              <View />
            </View>
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  undefined,
                  {
                    "paddingBottom": 8,
                    "paddingHorizontal": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "paddingHorizontal": 16,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        informal low
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 1
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        informal high
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 2
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        formal low
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 3
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          collapsable={false}
          onLayout={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(0, 0, 0, 0.12)",
              "borderRadius": 10,
              "elevation": 5,
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 5,
            }
          }
        >
          <View
            accessibilityState={
              {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                  },
                  {
                    "minHeight": 50,
                  },
                  {},
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "500",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontSize": 20,
                            "letterSpacing": 0.15,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          [
                            {
                              "minHeight": 30,
                              "paddingRight": 16,
                            },
                            {
                              "marginBottom": 2,
                            },
                            {
                              "color": "#3F51B5",
                              "fontSize": 16,
                              "fontWeight": "400",
                            },
                          ],
                        ],
                      ],
                    ]
                  }
                >
                  Declarative Future
                </Text>
              </View>
              <View />
            </View>
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  undefined,
                  {
                    "paddingBottom": 8,
                    "paddingHorizontal": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "paddingHorizontal": 16,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        informal low
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 3
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        informal high
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 4
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "marginBottom": 8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onPress={[Function]}
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        formal low
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        :
                      </Text>
                    </View>
                    <View
                      size={5}
                      style={
                        {
                          "flex": 5,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "fontSize": 20,
                            },
                          ]
                        }
                      >
                        conj 5
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
